[frontend] added teams and some small fixes

tzurE opened the pull request:
Frontend - 
added the teams add tab with edits for each team and everything
fixed issue where team members weren't showing in the team panel
fixed issue where spinner was stuck on the left. now it's in the middle.

please notice that there are a lot of changes because of all sorts of automatic changes that ember does to file system.
so you'll have a lot of stuff to go over on
I'm not sure if I can just commit the some specific files. I guess not because of the branches.
tell me if something needs fixing, it works well on my machine  
thanks.

Tzur-i committed:
[frontend] added teams and fixed issues
added the teams tab with edits and everything
fixed issue where team members weren't showing in the team panel
fixed issue where spinner was stuck on the left. not it's in the middle.

Tzur-i committed:
something went wrong, pulling and trying again
Merge branch 'master' of https://github.com/tzurE/rafee

holandes22 commented on Vagrantfile~:
```diff
@@ -0,0 +1,29 @@
+# -*- mode: ruby -*-
```
You should remove this swap file
Add an entry to the gitignore file to avoid this happening again

tzurE replied:
done

holandes22 commented on .vagrant/machines/default/virtualbox/action_provision:
```diff
@@ -0,0 +1 @@
+1.5:aaa5b7f4-2181-4b5f-9f45-f0429ceafe22
```
Why do we need this? this is vagrant metadata

tzurE replied:
the file itself? I have no idea. it's auto generated from vagrant I  think.

holandes22 replied:
I'm thinking it got commited by mistake, please remove it from the PR

tzurE replied:
hmm, whats the PR? delete it or just delete the file?

holandes22 replied:
Make a new commit in your for removing that file. No need for a new PR. same goes for pyc files

holandes22 commented on frontend/app/pods/admin/teams/team/edit/route.js:
```diff
@@ -0,0 +1,23 @@
+export default Ember.Route.extend(TeamRouteMixin, {
+
+  actions: {
+    //TODO: put this in a mixin
```
Can you do the TODO comment? this will be also used by other routes so it should be extracted in a mixin. The user.edit route uses it for example

tzurE replied:
just put it in a mixin, say like "Transition-mixin"
and just import it to every route?
no prob

holandes22 replied:
Exactly. Call the mixin file "alert-on-dirty-transition"

tzurE replied:
will be ready for the next pull-request I send, added it to my TODO tasks.

tzurE replied:
Done.

holandes22 commented on frontend/app/pods/admin/teams/template.hbs:
```diff
@@ -7,6 +7,7 @@
 <div class="row">
   <div class="small-5 columns">
     {{#each model as |team|}}
+      {{#if team.name}}
```
Is it a legal case to have a nameless team? I think it should be mandatory (if not the API needs to be modified)

tzurE replied:
I've actually added this here because sometimes it opens up a nameless team temporary, which lives only in the ember.store without it going to the db or something. I do this so there won't be any empty entries. if you take it off, like in the users section, each time you press "add" for a new user or team, and then go back - you see an empty entry. only after a refresh it goes away. 
but no, you can't have a nameless team. this was made only for these kind of situations.

holandes22 replied:
So there might be another problem. It shouln't be the way you describe it. 
This sort of hides that bug so I prefer to take it out and understand what is the underlying problem

tzurE replied:
from my understanding - the Amber.store keeps a copy if each new opened model before it calls the submit-action and does the save() function.
I'm not sure how to prevent this, as I'm fairly new to the concept of ember.
I'll try to read up on it, but it'll take some time. I'll add this as a fronend bug.

holandes22 commented on provisioning/inventory/hosts~:
```diff
@@ -0,0 +1,7 @@
+[devservers]
```
Same as before, swap file

tzurE replied:
done

holandes22 commented on provisioning/inventory/hosts:
```diff
@@ -1,6 +1,6 @@
 [devservers]
 # TODO: relative path to private key breaks ability to run vagrant up from anywhere in the project
-vagrant ansible_ssh_host=127.0.0.1 ansible_ssh_port=2222 ansible_ssh_user=vagrant ansible_ssh_private_key_file=./.vagrant/machines/default/virtualbox/private_key
+vagrant ansible_ssh_host=127.0.0.1 ansible_ssh_port=2222 ansible_ssh_user=vagrant
```
How is ansible being able to auth with no key or password? Are you positive this works ok?

tzurE replied:
I don't know how. maybe it uses my generic key that's in the default location?
it works great. every time I run the "vagrant provision" command, things work like it should, and I get a play recap and everything eventually. 
when i add this line - it just breaks down.

holandes22 replied:
Ok, I'll try it out in my laptop (it was working fine for me with that line). 
BTW, check my general comment on this. I prefer changes to the infrastructure to be done in a separate PR as in unrelated to the feature you implemented here

holandes22 commented:
How come the are a bunch of pyc files added to the PR?

holandes22 commented:
There are changes to ansible, vagrant and migrations. Can you clean those up and make them in an another PR? I prefer having unrelated stuff not mixed with the feature

holandes22 commented:
Needs some cleanup, but looks good, so two things:

1. Check the comments and try to clean the PR based on those
2. Add tests for at least the most basic stuff (you know, creating a team redirects to /teams, editing a team changes the values, etc)

tzurE commented:
1. not sure how to add tests, but ill add it to my to-do list.
2. how do I clean up the pull request? I mean,how do I change it once I did the request?
3. do I need to add a *.pyc rule to the .gitignore file?

Tzur-i committed:
added a alert-on-dirty mixin. some stuff for the .gitignore file, etc...

Tzur-i committed:
more gitignore stuff

tzurE committed:
Delete action_provision

tzurE committed:
Delete action_set_name

tzurE committed:
Delete id

Tzur-i committed:
deleted *.pyc and some other stuff

Tzur-i committed:
Merge branch 'master' of https://github.com/tzurE/rafee

Tzur-i committed:
ya

Tzur-i committed:
vagrant problem, must revert
Revert "Merge branch 'master' of https://github.com/tzurE/rafee"

This reverts commit 5fd285820ccfad6a3b1064e7715b838455da954d, reversing
changes made to dc1e4d4dd8dd208e50df6313ae834617ca38b7b1.

Tzur-i committed:
gitignore some more

Tzur-i committed:
added the Slideshows part of the GUI, some tests that aren't functional yet.

Tzur-i committed:
Merge branch 'master' of https://github.com/tzurE/rafee

Conflicts:
	.gitignore

tzurE committed:
Update .gitignore

Tzur-i committed:
ignore again

tzurE committed:
Delete action_set_name

Tzur-i committed:
added name & DS url in frontend templates.

Tzur-i committed:
slideshows - work with only 1 template and no auto refresh

Tzur-i committed:
everything works except slideshows with more than 1 template

Tzur-i committed:
sf

Tzur-i committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	frontend/package.json
